django/template/base.py

```diff
@@ -338,9 +339,10 @@ def split_contents(self):
 
 
 class Lexer:
-    def __init__(self, template_string):
+    def __init__(self, template_string, multiline=False):
         self.template_string = template_string
         self.verbatim = False
+        self.multiline = multiline
```

Comment:
Maybe it would make sense to store the regex locally:

```suggestion
        self.multiline = multiline
        self.tag_re = tag_re if multiline else tag_re_legacy
```

This way we wouldn't need to repeat this logic in the `DebugLexer` subclass.

Reply:
I like it. Sounds cleaner.